File: lib/core/auth/phone_auth_credentials.dart
import 'package:hilla_ride/core/constants/hilla_constants.dart';

/// Maps Iraqi phone numbers to Firebase Auth email identifiers (no OTP).
class PhoneAuthCredentials {
  PhoneAuthCredentials._();

  static String normalizePhone(String raw) {
    var digits = raw.replaceAll(RegExp(r'\D'), '');
    if (digits.startsWith('964')) {
      digits = digits.substring(3);
    }
    if (digits.startsWith('0')) {
      digits = digits.substring(1);
    }
    return '${HillaConstants.defaultCountryCode}$digits';
  }

  static String toAuthEmail(String phoneE164) {
    final digits = phoneE164.replaceAll(RegExp(r'\D'), '');
    return '$digits@hello-tiktok.app';
  }

  static bool isValidPassword(String password) => password.length >= 6;

  static bool isValidIraqiPhone(String raw) {
    var digits = raw.replaceAll(RegExp(r'\D'), '');
    if (digits.startsWith('964')) {
      digits = digits.substring(3);
    }
    if (digits.startsWith('0')) {
      digits = digits.substring(1);
    }
    return RegExp(r'^7\d{9}$').hasMatch(digits);
  }
}

File: hilla_ride_ios/HelloTukTuk/Core/Auth/PhoneAuthCredentials.swift
import Foundation

enum PhoneAuthCredentials {
    static let defaultCountryCode = "+964"
    static let authEmailDomain = "hello-tiktok.app"

    static func normalizePhone(_ raw: String) -> String {
        var normalized = raw
        let arabicIndic = Array("٠١٢٣٤٥٦٧٨٩")
        let easternArabic = Array("۰۱۲۳۴۵۶۷۸۹")
        for (index, digit) in arabicIndic.enumerated() {
            normalized = normalized.replacingOccurrences(of: String(digit), with: String(index))
        }
        for (index, digit) in easternArabic.enumerated() {
            normalized = normalized.replacingOccurrences(of: String(digit), with: String(index))
        }

        var digits = normalized.filter(\.isNumber)
        if digits.hasPrefix("964") {
            digits.removeFirst(3)
        }
        if digits.hasPrefix("0") {
            digits.removeFirst()
        }
        return "\(defaultCountryCode)\(digits)"
    }

    static func toAuthEmail(_ phoneE164: String) -> String {
        let digits = phoneE164.filter(\.isNumber)
        return "\(digits)@\(authEmailDomain)"
    }

    static func isValidPassword(_ password: String) -> Bool {
        password.count >= 6
    }

    static func isValidIraqiPhone(_ raw: String) -> Bool {
        var digits = normalizePhone(raw).filter(\.isNumber)
        if digits.hasPrefix("964") {
            digits.removeFirst(3)
        }
        if digits.hasPrefix("0") {
            digits.removeFirst()
        }
        let pattern = "^7\\d{9}$"
        return digits.range(of: pattern, options: .regularExpression) != nil
    }
}
